src/serena/agent.py

```diff
@@ -47,8 +47,12 @@ def __init__(self, config_dict: dict[str, Any], project_name: str, project_root:
         if project_root is None:
             project_root = Path(config_dict["project_root"])
         self.project_root: str = str(project_root.resolve())
-        self.ignored_dirs: list[str] = config_dict.get("ignored_dirs", [])
+        self.ignored_paths: list[str] = config_dict.get("ignored_paths", [])
```

Comment:
This is problematic. Users who don't update their configs will have this fail silently, so we need an exception if it's not there.

Reply:
I see that you (perhaps unexpectedly) handled backward compatbility below.